### DISCUSSION TITLE
Android TV?

### DISCUSSION THREAD

**Discussion by Alimond52** [GerryDazoo/Slinger#611] [General]
Hi, has anyone managed to get SlingerPlayer for Android working for an Android TV?
If so, which manufacturer and model?
Tips how to make it work?
I wish you a nice summer

**Comment by GerryDazoo** [comment]
You want to use the "Firestick" version (which is Android). The android version is for phones that can be rotated to switch to full-screen mode.

**Reply by Alimond52** [reply]
Hi Gerry,
I installed the "Firestick" version on my Android TV box Mecool. I was able to install the apk file using the file manager. And when I immediately open it, the Play window appears. But I cannot enter any IP address in that window. It says **http:/**/, then it's empty, and it's not possible to fill in either with a BT keyboard or wireless mouse. Both of these devices work on this Android TV box otherwise. But entering an IP address in SlingerPlayer does not work.
It is also not possible to **start** the SlingerPlayer app after it has been closed after installation. It is not in the App list at all. Maybe it's because I didn't do the "Side-loaded installation and then there won't be a shortcut to the startup file?" I see that there is a folder in the box where the SlingerPlayer is installed.
How well is this version tested against Android TV?
Thanks in advance.

**Reply by GerryDazoo** [reply]
Sorry for the complete lack of documentation on the app.
On the firestick remote there's the menu button ( three horizontal lines) that will bring to a page to edit the URLs. But that shouldn't be necessary is your TV  and the Server are on the same LAN. You are using the latest and greatest server 4.01?
I don't have an android TV so I haven't done any other testing on other devices. Maybe there is someone else here that has got it to work on their Android TV. But long story short because I don't have any way to remote debug issues there's not much I can do to address any issues you may find.

**Reply by GerryDazoo** [reply]
I'd try using ADB to install the APK.

**Reply by blasco403** [reply]
The 3 dashes button is called the "Menu" button. When the URL line is highlighted, press that button. Should take you to a screen to edit and then pop up a keyboard when you press "ok". Hit back/return and it will prompt to save.

On the nvidia shield, as another thread mentioned, you have to change the button mapping to send the menu button press to the application.

On my nvidia shield the app icon does not show up (something to do with no "menu" icon not being defined in the app). It does show up if you go into your installed applications and can be launched from there. I use an app called Mix-Apps the show icons for all installed apps. I have this as my only favorite.

Good luck.

**Reply by Alimond52** [reply]
Thank you, I will test later!!!

**Reply by Alimond52** [reply]
Hi Gerry and Blasco.
I've been testing for a few hours.
I don't have a Firestick myself. So I don't know much about it.
But from what I understand Firestick builds on Android OS as well?
But not sure if an app made for Firestick can also be installed and used on an "Android TV"?
The difference between apps for Andriod mobiles and Android TV is partly that the apps for Android TV are **landscape** to be displayed correctly in an Android TV. In addition, you **interact** differently, as you almost always use a simple **remote** control.
So the Menu button you refer to with three **dashes** is not on an Android TV remote control.
I have tried all file management apps like ADB and others. But you cannot access any files and therefore cannot edit them either.
So the only chance as I see it is to enter **manually** the Slinger server address in the interface for Slinger Player. But I only get one window, and it is not possible to put any characters in it, see the photo. But you can just close the window...
An Android TV box like this does not detect servers on the LAN network either.
However, a PC computer in the same LAN senses that there is a Slinger server correctly and installs the path perfectly.
For me, it would be great if your Firestick app could work. But I don't know how to proceed?
Some tips
Thanks!
![20230704_123900](https://github.com/GerryDazoo/Slinger/assets/123761450/149b3fad-452b-4c43-8a51-21350cf2cdcf)

**Reply by blasco403** [reply]
As I mentioned in my other post, I had the same issue on my nvidia shield. I had to customize the action of the "menu" button in the TV settings to pass the button to the application instead of bringing up the TV settings menu. Once I did that, I was able to press the "menu" button on remote and it worked as mentioned.

Look in your TV settings to see if you can map the same action to the button.

Good luck.

**Reply by Alimond52** [reply]
Thanks Blasco, I am trying to understand what you mean. My Android TV is not built into a TV. It is a separate Andorid TV box (Mecool), and a small remote. I am not sure if it possible to map??? I know that Nvidia is one of the best with everything.

**Reply by blasco403** [reply]
Ok, look in the settings of your mecool device to see if you can remap the "menu" button on your remote to send to the application instead of bringing up the settings on your mecool device.

**Reply by blasco403** [reply]
If you can't make it work on your mecool device. A Amazon fire stick works good for slingerplayer and other streaming services. They are cheaper than a shield as well. Amazon prime days are coming soon so they should be on sale. In Canada the fire stick 4k max is on sale right now for $39cad I think. I suggest you get the new 4k max even if your tv doesn't support 4k as the device is faster and has more memory and storage.

I have multiple fire stick lites that have minimal left over storage once I load Kodi/SlingerPlayer etc. I have to clean up cache/etc all the time to free up storage when upgrading apps/etc. Therefore why I suggest the 4k max - which is what I should have purchased.

I got a shield from work as a retirement gift, it works great but expensive for just streaming and not playing games for using as a plex server.

Good luck again.

**Reply by f520k510** [reply]
You can install "android tv remote app" on your android smartphone.  There are tons of android tv remote apps on play store.  I installed several of them and one of them has a menu button that works to open the url input field on the slingerplayer.

You need to pair the android tv remote app with your android tv box the first time.  Then you don't need to pair again.
Once you enter the urls of your slingbox, you don't need to use android tv remote app again.
Just use your physical remote.

**Reply by Alimond52** [reply]
Thanks Blasco, nice of you to help!!! I am soon going to the location abroad, there it is a real Android TV (Sony). I will try there. And perhaps the ordinary mobile Android version of Slinger Player can work in Landscape mode? I also would like to test "Android TV Remote" app for phone. Can you send me a link of the app that have a menu for open URL. Many thanks!!!

**Reply by blasco403** [reply]
As I understand, you will have to use the firestick version of the app on the Android TV. You will probably have to setup the menu key to send to app as I mentioned before so you can enter URL's.

The app I mentioned has nothing to do with URL's. It is called Mix-Apps, should be available on Play Store. The apps allows access to the SlingerPlayer app icon as it probably won't show up under the regular Menus once sideloaded. 

Good luck.

**Reply by f520k510** [reply]
@Alimond52 
If you refer to the Android tv remote that i mentioned, it's here https://play.google.com/store/apps/details?id=com.allrcs.RemoteForPanasonic


I worked not only on panasonic tv but also my fire tv.  

Maybe it will work on your sony tv as well.

**Reply by Alimond52** [reply]
Thank you so much! I have installed the remote app in my phone now and test it in a week!

**Reply by Alimond52** [reply]
Hello again,
I have tried the tips you were kind enough to give me.
So I'm at the other address and trying to get my Sony Android TV to be able to Configure URL addresses for Slinger Player for Firestick.
I have installed the Pansonic remote control on my mobile you mention. I have connected the app with my Sony TV. But then I don't know how to proceed. I have tried almost every option on the Pansonic app. There is so much advertising in that app that it is basically unusable...
I have also tested two wireless keyboards and a mouse, to be able to enter my URL address. But absolutely nothing happens as expected. If I press the number "1" on the keyboard, channel 1 starts in Tvn instead...
Usually with other apps where there is a field for the user to be able to input type text, numbers in, so the field is highlighted and possibly a cursor flashes. But here nothing like that happens at all. It just looks like it did in the picture I sent above.
I'm starting to doubt if the Slinger app really works for Android TV.
If you have any more idea please guide me bit by bit. Don't assume I understand your thinking in every way. Something that is obvious to you may not be to me. So describe step by step.
I would be extremely grateful if you would try to help me. So far nothing has worked. I feel lost after so many hours.

**Reply by f520k510** [reply]
You said you could connect the panasonic app with your sony tv.  Were you able to select the first panasonic remote and control your sony tv?  Because you need to enable developer's option to allow the panasonic app to work.  Assuming you can control your sony tv with the panasonic app, then all you have to do to make it work is use your SONY TV remote to open the slingerplayer app, move cursor to the url dropbox, and then press the menu button on the PANASONIC app.  You should then be able to add url to the list.  Once you are done, press back button on either the panasonic app or the sony tv remote.



Also, you are using the firestick version of slingerplayer, right?  If otherwise, it won't work on smart tv.

Here is a video showing on how to enable developer option:
https://www.youtube.com/watch?v=CQxOXqaO5og

If you want to connect the panasonic tv app with your sony tv, both devices must be in the same wifi or lan.
After you open the panasonic tv app, click connect.  It will bring up a list of android smart tv with its ip.  
You then choose your sony tv.  It will show a 6 digit code for you to enter on your tv.  
After entering it on your sony tv, you should be connected and be able to control your tv.
